### PR TITLE
[None][chore] BREAKING: Remove AutoDeploy from TensorRT-LLM

### PR DESCRIPTION
**Pull request by bmarimuthu-nv**
## Summary

- **BREAKING:** remove the in-tree AutoDeploy backend, examples, documentation, and tests, including backend selection in serving, the LLM API, checkpoint loading, and benchmarking
- remove dependent packaging, CI inventory, scanning, and ownership configuration
- address review feedback: preserve chunked-prefill validation coverage; remove unused loader/serve arguments, unreachable legacy logits and request-validation branches, unused V-stride overrides, and the orphaned non-update FLA implementation
- preserve the live FLA update path and default attention stride calculations
- rebase onto main at `fffdfba72`, retaining upstream OpenEngine, guided-decoding, performance, and speculative-decoding changes while removing their obsolete AutoDeploy references

## Validation

The deletion and review fixes at `a264233ad` passed the following local checks. The two subsequent MiniMax test-fixture repairs in `ef428052a` also pass all applicable file pre-commit checks:

- applicable pre-commit checks pass, including formatting, lint, YAML/TOML, generated configuration checks, and test inventory validation
- repository lightweight mypy passes for its 17 configured source files
- all 62 modified Python sources relative to main compile
- no tracked AutoDeploy identifiers or unresolved merge entries remain
- diff whitespace checks pass; commits have DCO sign-offs

## CI and test limitations

Pipeline #60745 completed both x86_64 and SBSA builds successfully. Its progress report exposed two stale MiniMax test fixtures: the shared-expert helper rejected `lora_params`, and the LoRA composition fixture lacked `routed_output_is_global`. Both fixtures are repaired in `ef428052a`, with existing numerical and LoRA-forwarding assertions preserved. The compliance scan separately aborted because its Jenkins agent failed to launch (jnlp exit 255).

CI with fail-fast disabled has been dispatched on `ef428052a`; runtime results remain pending. No failures have been waived.

The local environment lacks PyTorch and compiled TensorRT-LLM bindings. Runtime tests, bindings-aware type checking, golden-manifest verification, and C++/GPU validation therefore remain dependent on CI or a matching build environment.

## E2E output

No inference output generated locally.

### CODE REVIEW

**Review by govind-ramnarayan — APPROVED** [review]

File: tests/unittest/disaggregated/test_chunked_transfer.py [hunk 1]
```diff
@@ -631,53 +631,6 @@ def test_pipelined_transfer_requires_gen_first_flow():
         PyExecutor._validate_request(executor, request)
 
 
-def test_pipelined_transfer_requires_chunked_prefill_for_auto_deploy():
```
Comment (zhaoyangwang-nvidia) [inline tests/unittest/disaggregated/test_chunked_transfer.py:634]: These two tests were the only coverage for the `_enable_chunked_prefill` fallback in `PyExecutor._validate_request` (py_executor.py:5698-5703), which existed only for the AutoDeploy engine. After this PR nothing in the tree writes `_enable_chunked_prefill`, and `ModelEngine.__init__` always sets `attn_runtime_features`, so that branch is unreachable. Could it be collapsed to `self.model_engine.attn_runtime_features.chunked_prefill` with one renamed test kept, rather than leaving unreachable code with no coverage at all?

File: tensorrt_llm/_torch/pyexecutor/model_loader.py [hunk 2]
```diff
@@ -433,9 +433,6 @@ def _construct_checkpoint_loader(
             requested_checkpoint_io_policy,
             partial_model_loading,
         )
-    if backend == "_autodeploy":
-        return None
-
     from tensorrt_llm._torch.models.checkpoints.base_checkpoint_loader import \
```
Comment (zhaoyangwang-nvidia) [inline tensorrt_llm/_torch/pyexecutor/model_loader.py:431]: With the `_autodeploy` early return gone this function always returns a loader, so the `Optional` in the return annotation at line 425 can go. The `backend` parameter is now always `"pytorch"` as well (`TorchLlmArgs.backend` is `Literal["pytorch"]`), which makes the `backend != "pytorch"` branch in `_resolve_checkpoint_io_policy` at line 372 unreachable. Worth removing both, consistent with dropping `warn_non_pytorch_checkpoint_io_policy_fallback` in llm_args.py.

File: tensorrt_llm/commands/serve.py [hunk 3]
```diff
@@ -198,12 +198,7 @@ def is_non_default_or_required(param_name, value, backend, explicit_cli_keys):
            for s in cli_derived_fields.get(param_name, ())):
         return True
 
-    if backend == "_autodeploy":
-        from tensorrt_llm._torch.auto_deploy.llm_args import \
-            LlmArgs as AutoDeployLlmArgs
-        llm_args_class = AutoDeployLlmArgs
-    else:
-        llm_args_class = TorchLlmArgs
+    llm_args_class = TorchLlmArgs
```
Comment (zhaoyangwang-nvidia) [inline tensorrt_llm/commands/serve.py:215]: `backend` is now unused in this function. Please drop the parameter and update the call site at line 347 and `test_serve_is_non_default_or_required_helper`.

File: tensorrt_llm/llmapi/llm.py [hunk 4]
```diff
@@ -1494,7 +1482,7 @@ def _prepare_sampling_params(
             )
 
         # auto enable context and/or generation logits flags, as they are required by logprob computation for TRT backend.
-        if self.args.backend not in ["pytorch", "_autodeploy"]:
+        if self.args.backend != "pytorch":
```
Comment (zhaoyangwang-nvidia) [inline tensorrt_llm/llmapi/llm.py]: `TorchLlmArgs.backend` is `Literal["pytorch"]` and `BaseLLM.__init__` raises for any other value, so this condition is unconditionally false and the block below can never run. That leaves `SamplingParams._context_logits_auto_enabled` / `_generation_logits_auto_enabled` permanently `False` (the two assignments here are their only writers), which in turn makes the `and not self._..._auto_enabled` terms in `_need_return_context_logits` / `_need_return_generation_logits` (sampling_params.py:579,583) constant. Their own comment reads "Can be deprecated after migration to PyTorch backend" - can the whole chain be removed here instead of refreshing the dead condition?

File: tensorrt_llm/llmapi/llm.py [hunk 5]
```diff
@@ -1569,7 +1557,7 @@ def _add_whisper_suppress_tokens_logits_processor(
     def _check_arguments(self, prompt_len: int, sampling_params: SamplingParams,
                          is_gen_only: bool) -> None:
 
-        if self.args.backend in ["pytorch", "_autodeploy"]:
+        if self.args.backend == "pytorch":
             # Check prompt length against max_num_tokens to filter illegal requests.
             # Skip check for gen-only requests
             if self.args.backend == "pytorch" and not self.args.enable_chunked_prefill and not is_gen_only:
```
Comment (zhaoyangwang-nvidia) [inline tensorrt_llm/llmapi/llm.py]: The outer `if` now already narrows to `"pytorch"`, so this repeated `self.args.backend == "pytorch"` is always true; it only carried meaning while the outer check was `in ["pytorch", "_autodeploy"]`. Related, though it predates this PR: everything after the `return` in this branch reads `build_config` and the engine `config.json`, and is unreachable for the same reason.

File: cpp/tensorrt_llm/thop/attentionOp.cpp [hunk 6]
```diff
@@ -957,7 +957,7 @@ class Runner : public RunnerBase
                 enqueue_params.cu_kv_seqlens = cu_kv_seqlens->data_ptr<int32_t>();
             }
             // Pass V's actual token stride so the FMHA runner handles both
-            // contiguous V (AutoDeploy) and non-contiguous V (PyTorch backend
+            // contiguous and non-contiguous V layouts (PyTorch backend
```
Comment (zhaoyangwang-nvidia) [inline cpp/tensorrt_llm/thop/attentionOp.cpp]: The override this comment describes is never wired up: line 964 writes `EnqueueContextParams::v_stride_in_bytes`, but nothing reads it - `common/attentionOp.cpp` never assigns `fmhaParams.vStrideInBytes`, so `MHARunnerParams::vStrideInBytes` stays at its `= 0` default and the `if (runnerParams.vStrideInBytes > 0)` branch at `fmhaRunner.cpp:193` is unreachable. This came in with the AutoDeploy MLA backend, so could it be removed here (the field, the `if`, and this write)? Rewording alone leaves the comment claiming a general PyTorch-backend capability that no call path can exercise.

File: tests/unittest/auto_deploy/singlegpu/custom_ops/fla/test_fla_cached_gated_delta_rule.py [hunk 7]
```diff
@@ -1,402 +0,0 @@
...
-
-import pytest
-import torch
-import torch.nn.functional as F
-
-# Register all auto_deploy custom ops
-import tensorrt_llm._torch.auto_deploy.custom_ops  # noqa: F401
-from tensorrt_llm._torch.auto_deploy.custom_ops.attention_interface import BatchInfo
-from tensorrt_llm._torch.modules.fla.chunk import chunk_gated_delta_rule
-from tensorrt_llm._torch.modules.fla.fused_recurrent import fused_recurrent_gated_delta_rule_fwd
```
Comment (zhaoyangwang-nvidia) [inline tests/unittest/auto_deploy/singlegpu/custom_ops/fla/test_fla_cached_gated_delta_rule.py:41]: This import was the only external consumer of `fused_recurrent_gated_delta_rule_fwd`. With this test deleted, the non-`_update` half of `tensorrt_llm/_torch/modules/fla/fused_recurrent.py` has no entry point left: the kernel at line 21, `_fwd` at 119, `FusedRecurrentFunction` at 177 and `fused_recurrent_gated_delta_rule` at 220 are only reachable from each other, `fla/__init__.py` is empty, and there is no `torch.library` registration. The `_update` family from line 325 is still live via `gdn_mixer.py`, so only the dead half should go.

**Review by StanleySun639 — APPROVED** [review]

**Review by nv-guomingz — APPROVED** [review]
LGTM

**Review by Shixiaowei02 — APPROVED** [review]

**Review by tburt-nv — APPROVED** [review]

**Review by zhaoyangwang-nvidia — APPROVED** [review]

**Review by Barry-Delaney — APPROVED** [review]
LGTM for the MoE scope: no MoE backend/scheduler code is touched and no non-AutoDeploy module imported the removed MoE custom ops.

**Review by QiJune — APPROVED** [review]
LGTM

**Review by xinhe-nv — APPROVED** [review]

**Review by nvpohanh — APPROVED** [review]

**Review by longlee0622 — APPROVED** [review]

**Review by ruodil — APPROVED** [review]
